src/components/MultiColumnMeasures.tsx

```diff
@@ -0,0 +1,258 @@
+import AddIcon from '@mui/icons-material/Add';
+import { Fab, Card, CardContent, CardHeader, Typography } from '@mui/material';
+import { useTheme } from '@mui/material/styles';
+import { useState } from 'react';
+import { v4 as uuidv4 } from 'uuid';
+import assessmentTerms from '../assets/assessmentTerms.json';
+import { usePagination } from '../hooks';
+import useDataStore from '../stores/data';
+import { Term, TermCard } from '../utils/types';
+import MultiColumMeasuresCard from './MultiColumnMeasuresCard';
+
+interface MultiColumnMeasuresProps {
+  generateID?: () => string;
+  terms?: Term[];
+}
+
+const defaultProps = {
+  generateID: uuidv4,
+  terms: assessmentTerms,
+};
+
+function MultiColumnMeasures({
+  generateID = uuidv4,
+  terms = assessmentTerms,
+}: MultiColumnMeasuresProps) {
+  const theme = useTheme();
+  const columns = useDataStore((state) => state.columns);
+  const updateColumnIsPartOf = useDataStore((state) => state.updateColumnIsPartOf);
+  const assessmentToolConfig = useDataStore.getState().standardizedVariables['Assessment Tool'];
+
+  const assessmentToolColumns = Object.entries(columns)
+    .filter(
+      ([_, column]) => column.standardizedVariable?.identifier === assessmentToolConfig.identifier
+    )
+    .map(([id, column]) => ({ id, header: column.header }));
+
+  // Initialize term cards based on existing isPartOf relationships
+  const initializeTermCards = (): TermCard[] => {
+    const assessmentColumns = assessmentToolColumns.reduce((acc, { id }) => {
+      const column = columns[id];
+      if (!column.isPartOf) return acc;
+
+      const term = terms.find((t) => t.identifier === column.isPartOf?.termURL);
+      if (!term) return acc;
+
+      // Find or create card for this term
+      let card = acc.find((c) => c.term?.identifier === term.identifier);
+      if (!card) {
+        card = {
+          id: generateID(),
+          term,
+          mappedColumns: [],
+        };
+        acc.push(card);
+      }
+
+      // Add column to mappedColumns if not already present
+      if (!card.mappedColumns.includes(id)) {
+        card.mappedColumns.push(id);
+      }
+
+      return acc;
+    }, [] as TermCard[]);
+
+    return assessmentColumns.length > 0
+      ? assessmentColumns
+      : [{ id: generateID(), term: null, mappedColumns: [] }];
+  };
+
```

Comment:
This is a pretty complex piece of code. As far as I can tell, the reason you need to do all this is because you are inferring the existence of a "created" AssessmentTool from the mapping output in the `column`. I.e. instead of asking: how many tools has the user selected to map columns to? You are asking: what tools have columns been mapped to - and then you take the unique set of that. 

I don't think this is necessary and I also don't think it's intuitive. I.e. the action of "create new AssessmentTool" to annotate should have an effect in the state. And the action of mapping a column to one of these tools should have another effect in the state.

Reply:
> I don't think this is necessary and I also don't think it's intuitive. I.e. the action of "create new AssessmentTool" to annotate should have an effect in the state. And the action of mapping a column to one of these tools should have another effect in the state.

I agree and it does but not when the state has been set from re-uploading a data dictionary.